main.go

```diff
@@ -182,3 +193,41 @@ func isClusterPolicyAvailable() wait.ConditionFunc {
 		return !unavailable, nil
 	}
 }
+
+func jumpstartAddon(client client.Client) error {
+	gpuAddon := &nvidiav1alpha1.GPUAddon{}
+	err := client.Get(context.TODO(), types.NamespacedName{
+		Name:      common.GlobalConfig.AddonID,
+		Namespace: common.GlobalConfig.AddonNamespace,
+	}, gpuAddon)
+	isNotFound := k8serrors.IsNotFound(err)
+	if err != nil && !isNotFound {
+		return fmt.Errorf("failed to fetch GPUAddon CR %s in %s: %w", common.GlobalConfig.AddonID, common.GlobalConfig.AddonNamespace, err)
+	}
+	if isNotFound {
+		gpuAddon = &nvidiav1alpha1.GPUAddon{
```

Comment:
There is no need to initialize the GPUAddon struct here, we can just add the essential `ObjectMeta` fields and all the rest can be specified in the mutate function arg of the `CreateOrPatch`. For example:

```go
gpuAddon.ObjectMeta = metav1.ObjectMeta{
    Name:           common.GlobalConfig.AddonID,
    Namespace: common.GlobalConfig.AddonNamespace,
}

resp, err := controllerutil.CreateOrPatch(context.TODO(), client, gpuAddon, func() error {
		versionLabel := fmt.Sprintf("%v-version", common.GlobalConfig.AddonLabel)

		gpuAddon.ObjectMeta.Labels = map[string]string{
		    versionLabel: version.Version(),
		}

		gpuAddon.Spec = nvidiav1alpha1.GPUAddonSpec{
		    ConsolePluginEnabled: true,
		}

		return nil
	})


```

Reply:
I see your point. And will change, but I do not wish to override the Spec on update. i.e if the console plugin was turned off - it should stay that way.

Thanks for the input.